src/plugins/record_playback_plugin/playback_action.cc

```diff
@@ -163,13 +163,16 @@ void PlaybackAction::Initialize(YAML::Node options_node) {
   AIMRT_CHECK_ERROR_THROW(!metadata_.files.empty(),
                           "Empty bag! bag path: {}", options_.bag_path);
 
+  auto corrected_metadata_files = std::deque<MetaData::FileMeta>();
   for (auto& item : metadata_.files) {
     const auto db_file_path = std::filesystem::path(options_.bag_path) / item.path;
-
-    AIMRT_CHECK_ERROR_THROW(
-        std::filesystem::exists(db_file_path) && std::filesystem::is_regular_file(db_file_path),
-        "Can not find bag file '{}' in bag path '{}'.", db_file_path.string(), options_.bag_path);
+    if (!std::filesystem::exists(db_file_path) || !std::filesystem::is_regular_file(db_file_path)) {
+      AIMRT_WARN("Can not find bag file '{}' in bag path '{}', this file will be ignored.", db_file_path.string(), options_.bag_path);
+      continue;
+    }
+    corrected_metadata_files.emplace_back(std::move(item));
   }
+  metadata_.files = std::move(corrected_metadata_files);
```

Comment:
Consider using the erase-remove idiom to simplify this code

```
void processMetadataFiles() {
    auto is_valid_file = [](const std::filesystem::path& path) {
        return std::filesystem::exists(path) && std::filesystem::is_regular_file(path);
    };

    metadata_.files.erase(
        std::remove_if(metadata_.files.begin(), metadata_.files.end(),
            [&](const MetaData::FileMeta& item) {
                const auto db_file_path = std::filesystem::path(options_.bag_path) / item.path;
                if (!is_valid_file(db_file_path)) {
                    AIMRT_WARN("Can not find bag file '{}' in bag path '{}', this file will be ignored.", db_file_path.string(), options_.bag_path);
                    return true; // Remove this item
                }
                return false; // Keep this item
            }),
        metadata_.files.end());
}
```

Reply:
Maybe erase if doesn't work in gcc11.4.   https://godbolt.org/z/x9xf9e6xh